### ISSUE TITLE
Unable to run functions in any other namespace than default

### ISSUE THREAD

**Issue by erikedling** [serverless/serverless-kubeless#188]
So I'm just trying out kubeless through serverless and I'm currently running kubeless in my own namespace.  I'm using the env `KUBELESS_NAMESPACE `so that the config-map can be found. If I'm deploying the functions in the default namespace, it works fine. However, if I add the field "namespace" in the serverless.yml file in order to deploy the functions in my own namespace as well like so

```  
{
  provider:
    name: kubeless
    namespace: mynamespace
    runtime: nodejs6

  etc
}
```

And I run `serverless invoke --function capitalize --data '"WELCOME TO KUBELESS!"' -l`. I will get `TypeError: opts.log is not a function` in the file `serverless-kubeless/lib/invoke.js:128:16`.

If I log the "servicesInfo" variable in the same scope I get the following
```
{ 
  kind: 'ServiceList',
  apiVersion: 'v1',
  metadata: { 
    selfLink: '/api/v1/namespaces/default/services',
    resourceVersion: '129499598' },
    items: [ { metadata: [Object], spec: [Object], status: [Object] } ] 
}

```
I can see that it uses the default namespace and not the one specified in the serverless.yml file. Am I doing something completely wrong or Is this maybe not supported or is it a bug?

**Comment by erikedling** [comment]
Sorry about that but yes, when i set the "namespace" field under "provider" in serverless.yml it works to deploy in the specified namespace but not invoking the functions as it looks for the functions in the default namespace instead of the specified one.

Through kubeless CLI i can invoke with the following command:
`kubeless function call capitalize -d "WELCOME TO KUBELESS!" --namespace mynamespace`.

If I set the namespace field in connectionOptions to the value from the namespace variable(existing in the scope of the "invoke" method) it works perfectly.

**Comment by erikedling** [comment]
I could do the easy fix changing the field in connectionOptions after it has been created in the invoke method however I would much rather have wanted to make sure helpers.getConnectionOptions does it correctly in the first place. Unfortunately I can't seem to get a local minikube instance up and running on my personal computer to test things and I don't have too much time fixing that. I'm sorry, could someone else maybe do the simple fix?

**Comment by cscetbon** [comment]
@erikedling any progress on that issue ?

**Comment by arditdine** [comment]
I had the same issue but I noticed that when you create a new serverless project with the command:
`serverless create --template kubeless-nodejs --path new-project` works fine, you can deploy in every namespace and can invoke the functions.
If you check in package.json it uses serverless-kubeless v0.4.0. I've tested and everything works fine up to version 0.6.0
In the new versions of serverless-kubeless it doesn't work. The only difference I found between versions was that in invoke.js on line 115 the function was changed from `core.services.get` => `core.ns.services.get`
If you remove `ns` it works fine again.

**Comment by cscetbon** [comment]
@arditdine interesting ! I’ll have to try it

**Comment by arditdine** [comment]
@andresmgot I think it's safe to close this right?

**Comment by andresmgot** [comment]
Indeed, thanks @arditdine! This has been fixed in 0.11.1.
